app/javascript/packs/brave_rewards_banner_display.jsx

```diff
@@ -0,0 +1,417 @@
+      hiddenDonationAmounts.type = "hidden"
+      hiddenDonationAmounts.style.display = 'none';
+      hiddenDonationAmounts.onchange = this.handleDonationAmountsChange;
+      document.body.appendChild(hiddenDonationAmounts);
+
+      // Editable for tokens
+      for (let element of document.getElementsByClassName("sc-brqgnP")) {
+        element.setAttribute("contenteditable", true);
+        var observer = new MutationObserver(function(mutations) {
+          mutations.forEach(function(mutation) {
+            if (mutation.type == "contentList") {
+              return;
+            }
+            // TODO: (Albert Wang) Make sure the input are valid numbers
+            var donationAmounts = [];
+            for (let amountSpan of document.getElementsByClassName("sc-brqgnP")) {
+              donationAmounts.push(parseInt(amountSpan.textContent));
+            }
+            document.getElementById("donation-amounts-input").value = donationAmounts;
+            document.getElementById("donation-amounts-input").onchange();
+          });
+        });
+        // configuration of the observer:
+        var config = { characterData: true, attributes: false, childList: true, subtree: true };
+        // pass in the target node, as well as the observer options
+        observer.observe(element, config);
+      };
+
+      // Hide X-mark
+      document.getElementsByClassName("sc-bZQynM")[0].style = "display: none";
+    }
+  }
+
+  submitById(id, suffix) {
+    const url = '/publishers/' + this.props.publisher_id + "/site_banners/update_" + suffix;
+    var file = document.getElementById(id);
+    var reader = new FileReader();
+
+    // Don't upload if user didn't upload a new image
+    if (file.value == "" || file.value == null) {
+      return;
+    }
+    reader.readAsDataURL(file.files[0]);
+    reader.onloadend = function () {
+      const body = new FormData();
+      body.append('image', reader.result);
+      fetch(url, {
+        method: 'POST',
+        headers: {
+          'Accept': 'text/html',
+          'X-Requested-With': 'XMLHttpRequest',
+          'X-CSRF-Token': document.head.querySelector("[name=csrf-token]").content
+        },
+        credentials: "same-origin",
+        body: body
+      });
+    };
+  }
+
+  updateDescription(event) {
+    this.setState({description: event.target.value})
+  }
+
+  updateTwitch(event) {
+    let temp = this.state.socialLinks
+    temp.twitch = event.target.value
+    this.setState({socialLink : temp});
+  }
+
+  updateYoutube(event) {
+    let temp = this.state.socialLinks
+    temp.youtube = event.target.value
+    this.setState({socialLink : temp});
+  }
+
+  updateTwitter(event) {
+    let temp = this.state.socialLinks
+    temp.twitter = event.target.value
+    this.setState({socialLink : temp});
+  }
+
+  handleSubmit(event) {
+    const url = '/publishers/' + this.props.publisher_id + "/site_banners";
+    var request = new XMLHttpRequest();
```

Comment:
this appears to be unused? (you can add a js lint command in travis to catch these things)